20240328-queries-with-usersets.md

```diff
@@ -0,0 +1,176 @@
+# Meta
+[meta]: #meta
+- Name: Queries with usersets
+- Start Date: 2024-03-28
+- Author(s): @miparnisari
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- Relevant Issues: n/a
+- Supersedes: n/a
+
+# Summary
+[summary]: #summary
+
+This is a proposal to change the outcome of query APIs (specifically, Check and ListObjects APIs) when the target of the query is a userset. The proposal also involves adapting ListUsers API to return responses consistent with Check and ListObjects.
+
+# Definitions
+[definitions]: #definitions
+
+- **Object**: is of the form `objectType:objectId`.
+- **User**: is a specific object, such as `employee:maria` or `document:1`.
```

Comment:
This definition goes against our documentation.

A user is an object, a userset or everyone in a type with `objectType:*`

Reply:
Oh my intention was not to deviate :) i will rephrase

Reply:
PS how do we not have "what is a userset?" in the docs? 🤔

Reply:
It's also mentioned under https://openfga.dev/docs/concepts#what-is-a-user

Reply:
oh, my bad 🤦‍♀️